subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/AbstractFingerprintingStrategy.java

```diff
@@ -37,4 +51,29 @@ public String getIdentifier() {
     public CurrentFileCollectionFingerprint getEmptyFingerprint() {
         return emptyFingerprint;
     }
+
+    @Nullable
+    protected HashCode getNormalizedContentHash(FileSystemLocationSnapshot snapshot, FileSystemLocationSnapshotHasher normalizedContentHasher) {
+        try {
+            return normalizedContentHasher.hash(snapshot);
+        } catch (IOException e) {
+            throw new UncheckedIOException(failedToNormalize(snapshot), e);
+        } catch (UncheckedIOException e) {
+            throw new UncheckedIOException(failedToNormalize(snapshot), e.getCause());
+        }
```

Comment:
Yeah, contextual is not visible from this project, unfortunately.

I didn't catch other runtime exceptions mostly because we normally wouldn't catch those.  I wouldn't normally catch `UncheckedIOException` here, either, except that I know we're wrapping normal `IOException` instances coming from `IOSupplier.wrap()`.  I'm open to just catching `Throwable` here and adding some meaningful context if you think that makes more sense and feels less awkward.

Whatever improvements we make here, we'll do in a follow up PR.